pynbody/snapshot/grafic.py

```diff
@@ -196,27 +199,32 @@ def _read_pvar(self):
 
     def _read_grafic_file(self, filename, target_buffer, data_type):
         with FortranFile(filename) as f:
-            h = f.read_field(genic_header)
-            length = self._dlen * data_type.itemsize
-            alen = f.get_raw_memmapped(util._head_type)
-            if alen != length:
-                raise IOError("Unexpected FORTRAN block length %d!=%d" % (alen, length))
-            for readlen, buf_index, mem_index in (self._load_control.iterate_with_interrupts(family.dm, family.dm,
-                                                                                             np.arange(
-                                                                                                 1, h['nz']) * (
-                                                                                                 h['nx'] * h['ny']),
-                                                                                             functools.partial(
-                                                                                                 _midway_fortran_skip, f,
-                                                                                                 length))):
-
-                if buf_index is not None:
-                    re = f.get_raw_memmapped(data_type, readlen)
-                    target_buffer[mem_index] = re[buf_index]
-                else:
-                    f.seek(data_type.itemsize * readlen)
-            alen = f.get_raw_memmapped(util._head_type)
-            if alen != length:
-                raise IOError("Unexpected FORTRAN block length (tail) %d!=%d" % (alen, length))
+            h = {k: v for k, v in zip(genic_header['keys'], f.read_vector(genic_header['dtype'])[0])}
+            ii = 0
+            for _ in range(h['nz']):
```

Comment:
In its current form this breaks partial loading in Grafic (i.e. ability to load only part of the grid, which is provided by the LoadControl object)

Reply:
Actually the code as is seems to be working just fine, see the test grafic file.

Reply:
Unfortunately not - the test Grafic file is only 4^3=64, so the test as currently implemented always selects all particles!

If you update line 41 to `for test_len in [16, 32, 48]:`, for example, you see the problem